src/queries.rs

```diff
@@ -23,8 +34,28 @@ pub fn find_queries_to_cache(
             AND digest_text NOT LIKE '%?=?%'
             AND s.sum_rows_sent > 0
             AND q.rule_id IS NULL
-            ORDER BY s.sum_rows_sent DESC",
-            config.source_hostgroup, config.readyset_user
+            ORDER BY s.sum_rows_sent DESC
+            LIMIT {}",
+            config.source_hostgroup,
+            config.readyset_user,
+            (config.number_of_queries * 2)
```

Comment:
Possibly a stupid question, but why 2x? Just to have extras in case some can't be cached?

Reply:
No stupid. This is indeed to account for query not supported. We get twice as much queries we should cache and start testing them, stopping when we reach `number_of_queries` config.